src/components/e-date.vue

```diff
@@ -0,0 +1,326 @@
+            this.dateValue = date;
+          }
+        },
+        get() {
+          return this.invalid
+            ? this.textValue
+            : this.dateValue.format(this.format);
+        },
+      },
+    },
+    watch: {
+      /**
+       * Reset the invalid state when the date updates (most likely from the date field).
+       *
+       * @param {Object} newValue - The updated value.
+       * @param {Object} oldValue - The previous value.
+       */
+      dateValue(newValue, oldValue) {
+        if (newValue.toDate().getTime() !== oldValue.toDate().getTime()) {
+          this.invalid = false;
+
+          /**
+           * Sends the current value to the parent.
+           *
+           * @event change
+           *
+           * @property {Date} date - The current date value.
+           */
+          this.$emit('change', this.dateValue.toDate());
+        }
+      },
+
+      /**
+       * Handles updates on the value prop.
+       *
+       * @param {Date} date - The new date value.
+       */
+      value(date) {
+        this.dateValue = this.$dayjs(date);
+      },
+    },
+
+    // beforeCreate() {},
+    // created() {},
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeDestroy() {},
+    // destroyed() {},
+
+    methods: {
+      /**
+       * Event handler for 'enter' keypress on the input.
+       *
+       * @param {Event} event - The original DOM event.
+       */
+      onEnter(event) {
+        event.preventDefault(); // Prevents auto submit of form.
+        event.target.blur(); // Triggers update.
+      },
+
+      /**
+       * Event handler for the focus event of the input.
+       *
+       * @param {Event} event - The original DOM event.
+       */
+      onFocus(event) {
+        /**
+         * Forwards the focus event to the parent component.
+         *
+         * @event focus
+         *
+         * @property {Event} event - The original DOM event.
+         */
+        this.$emit('focus', event);
+      },
+
+      /**
+       * Event handler for the blur event of the input.
+       *
+       * @param {Event} event - The original DOM event.
+       */
+      onBlur(event) {
+        /**
+         * Forwards the blur event to the parent component.
+         *
+         * @event blur
+         *
+         * @property {Event} event - The original DOM event.
+         */
+        this.$emit('blur', event);
+      },
+    },
+    // render() {},
+  };
+</script>
+
+<style lang="scss">
+  @use '../setup/scss/variables';
+  @use '../setup/scss/mixins';
+  @use '../setup/scss/extends';
+  @use '../setup/scss/functions';
+
+  .e-date {
+    $height: 48px;
+    $border: 1px;
+
+    display: flex;
+    align-items: center;
+    padding: rem(0 variables.$spacing--25); // Equalizes vertical centering
+    border: $border solid transparent;
+    border-radius: 99999px; // Force half circles.
```

Comment:
todo: please remove this border radius but also apply a basic styling as for the other fields is used (light gray border, slight border radius).